### PR TITLE
Add helper trait to provide type-hinting for zend-view

### PR DESCRIPTION
**Pull request by thexpand**
In favor of the following proposal:
https://github.com/zendframework/zend-view/issues/102

### CODE REVIEW

File: src/View/HelperTrait.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
+<?php
+
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: Doc block with copyright and licence is missing. An example can be found here:

https://github.com/zendframework/maintainers/blob/abab4a99f3e6d1fac4d7400232ce30ee0bc95249/template/src/ConfigProvider.php#L2-L6

File: src/View/HelperTrait.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+/**
+ * Trait HelperTrait
+ *
+ * The trait provides convenience methods for view helpers,
+ * defined by the zend-form component. It is designed to be used
+ * for type-hinting $this variable inside zend-view templates via doc blocks.
+ *
+ * The base class is PhpRenderer, followed by the helper trait from
+ * the zend-form component. However, multiple helper traits from different
+ * Zend components can be chained afterwards.
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: "Zend components" is wrong, these are "Zend Framework components".

File: src/View/HelperTrait.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+use Zend\Form\View\Helper\FormTel;
+use Zend\Form\View\Helper\FormText;
+use Zend\Form\View\Helper\FormTextarea;
+use Zend\Form\View\Helper\FormTime;
+use Zend\Form\View\Helper\FormUrl;
+use Zend\Form\View\Helper\FormWeek;
+
+// @codingStandardsIgnoreStart
+/**
+ * Trait HelperTrait
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: "Trait HelperTrait" – what is this?! 😉
Can we find a short and meaningful name / description?

File: src/View/HelperTrait.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+use Zend\Form\View\Helper\FormSelect;
+use Zend\Form\View\Helper\FormSubmit;
+use Zend\Form\View\Helper\FormTel;
+use Zend\Form\View\Helper\FormText;
+use Zend\Form\View\Helper\FormTextarea;
+use Zend\Form\View\Helper\FormTime;
+use Zend\Form\View\Helper\FormUrl;
+use Zend\Form\View\Helper\FormWeek;
+
+// @codingStandardsIgnoreStart
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: Here is the start, where is the end?

File: src/View/HelperTrait.php [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+ * for type-hinting $this variable inside zend-view templates via doc blocks.
+ *
+ * The base class is PhpRenderer, followed by the helper trait from
+ * the zend-form component. However, multiple helper traits from different
+ * Zend components can be chained afterwards.
+ *
+ * @example @var \Zend\View\Renderer\PhpRenderer|\Zend\Form\View\HelperTrait $this
+ *
+ * @method string|Form form(FormInterface $form = null)
+ * @method string|FormButton formButton(ElementInterface $element = null, $buttonContent = null)
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: We can add also the data types. See my comment here: https://github.com/zendframework/zend-i18n/pull/102#discussion_r215139655

**Comment by froschdesign** [comment]
The usage of this trait must be explained in the documentation.

**Comment by froschdesign** [comment]
Btw. Please use the `develop` branch for new features.

https://github.com/zendframework/zend-form/blob/master/docs/CONTRIBUTING.md#what-branch-to-issue-the-pull-request-against

**Comment by thexpand** [comment]
@froschdesign 
I suggest adding the docs below the `View Helpers > AbstractHelper` section.
More information about the naming: https://github.com/zendframework/zend-i18n/pull/102#issuecomment-418711168

**Comment by froschdesign** [comment]
@thexpand 
Why not on the introduction page of the view helpers?

https://docs.zendframework.com/zend-form/helper/intro/

File: src/View/HelperTrait.php [hunk 6]
```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-form for the canonical source repository
+ * @copyright Copyright (c) 2005-2018 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: Please use for new files only the current year.

**Comment by weierophinney** [comment]
Thanks, @thexpand!

**Pull request merged**